docs/source/tutorials/hospital_admissions_model.qmd

```diff
@@ -499,14 +490,16 @@ class DayOfWeekEffect(metaclass.RandomVariable):
             sample_shape=(7,),
         )
 
-        return jnp.tile(ans, self.nweeks)[: self.len]
+        return (metaclass.TimeArray(
+            jnp.tile(ans, 100)[: n_timepoints]),
```

Comment:
I haven't been able to figure out why using `int(jnp.ceil(n_timepoints/7))` instead of hardcoding 100 doesn't work. I understand `jax` usually doesn't like that pattern, but the periodic effect works (see [here](https://github.com/CDCgov/multisignal-epi-inference/blob/ce055afcaa2998d152ce39b3efd248d98eb3ddef/model/src/pyrenew/process/periodiceffect.py#L64-L68)). Pre-specifying the number of weeks this effect needs to be tiled beforehand is not an option as it won't work with the posterior predictive method.

Reply:
In what way doesn't it work? Do you get an error?

Reply:
I'd suggest staying away from this pattern entirely. I'd have the returned array be of length 7 and then use a mask rather than a tiling to create an array aligned with the actual data. In other words, if I have some vector of dow (here a scaled simplex):
```python
dow_effects = 7 * jnp.array([0.1, 0.2, 0.1, 0.2, 0.1, 0.2, 0.1])
```

And I have a vector of values to which I'd like to associate the d.o.w. effect, which might be more or fewer than 7 entries long:

```python
pre_dow_effect_values_shorter = jnp.array([15, 7, 2, 4, 6])
pre_dow_effect_values_longer = jnp.array([35, 32, 13, 5, 10, 15, 36, 20, 3])
```

I do this with a lookup mask, e.g.:

```python
dow_index_shorter = jnp.array([5, 6, 0, 1, 2])
dow_index_longer = jnp.array([3, 4, 5, 6, 0, 1, 2, 3, 4])
```

I can then do:

```python
dow_effects_shorter = dow_effects[dow_index_shorter]
dow_effects_longer = dow_effects[dow_index_longer]
```

Reply:
> In what way doesn't it work? Do you get an error?

That's right, @damonbayer. The compiler doesn't like casting floats into ints in some situations. Not always the case, as the periodic effects work just fine. Will work on this next week.

Reply:
> > In what way doesn't it work? Do you get an error?
> 
> That's right, @damonbayer. The compiler doesn't like casting floats into ints in some situations. Not always the case, as the periodic effects work just fine. Will work on this next week.

Did you try integer division? https://stackoverflow.com/a/17511341

Reply:
I'll try that, @damonbayer. Thanks!

Reply:
Works like a charm, @damonbayer! Thanks.